src/promptflow/promptflow/_sdk/entities/_flow.py

```diff
@@ -190,3 +193,17 @@ def outputs(self):
         return {k: v.type.value for k, v in self._executable.outputs.items()}
 
     # endregion
+
+    def __call__(self, *args, **kwargs):
+        # run flow test here
+        # TODO: cache the submitter?
+        from promptflow import PFClient
+        from promptflow._sdk.operations._test_submitter import TestSubmitter
+
+        if args:
+            raise UserErrorException("Flow can only be called with keyword arguments.")
+        client = PFClient()
+        # TODO: gap: variants, environment_variables
+        result = client.flows._test_flow(flow=self, inputs=kwargs, variant=self.variant)
```

Comment:
I think we should leverage FlowInvoker not directly use flow test, as we don't need stuff like the output data persist logic

Reply:
Offline synced, we need to decide whether to put connection/variant resolve logic in test submitter or in flow invoker.